dartagnan/src/main/java/com/dat3m/dartagnan/encoding/ExpressionEncoder.java

```diff
@@ -304,11 +243,13 @@ public Formula visit(IExprBin iBin) {
 
     @Override
     public Formula visit(IExprUn iUn) {
-        //TODO if inner is a boolean formula, we still want to encode it as boolean in case of conversions
-        Formula inner = encodeAsInteger(iUn.getInner());
+        Formula inner = encode(iUn.getInner());
         Type targetType = iUn.getType();
         switch (iUn.getOp()) {
             case MINUS -> {
+                if (inner instanceof BooleanFormula bool) {
+                    return bool;
+                }
```

Comment:
Kinda fishy. Are there cases where this code actually runs, i.e., do we ever produce somthing like `-false`?